### DISCUSSION TITLE
Don't allow choices with same value in dropdown

### DISCUSSION THREAD

**Discussion by Pitcentr** [directus/directus#13509] [Feature Requests]
### Describe the Bug

When you create a multiple list with the same values, an error occurs, selecting one selects all the others

### To Reproduce

https://recordit.co/eLAkM8n9Hm

![изображение](https://user-images.githubusercontent.com/34168062/168827554-5f8a963f-6f50-4e30-86f2-71f74f853737.png)

![изображение](https://user-images.githubusercontent.com/34168062/168827593-4ec51d56-50d0-4cff-acee-3d3ad5c1a3a1.png)
![изображение](https://user-images.githubusercontent.com/34168062/168827742-f9b531d8-2160-4077-b00e-69268ff09b0c.png)
![изображение](https://user-images.githubusercontent.com/34168062/168827769-d921f490-19a2-4880-b081-97d7a5ce1bdf.png)

### What version of Directus are you using?

9.10.0

### What version of Node.js are you using?

v14.19.1

### What database are you using?

mariadb

### What browser are you using?

Firefox

### How are you deploying Directus?

npm

**Comment by azrikahar** [comment]
Is there a particular reason to have multiple choices with the same value? If it's intentional, I personally wouldn't categorize this is a bug or error, but I do agree it can be improved in the sense that duplicate value should be prevented (or maybe overrides the previous value).

**Comment by Pitcentr** [comment]
you need for the opening times by day of the week, 
0 - Monday 8:00-19:00 
1 - Tuesday 8:00-19:00 
.... 
5 - Saturday 9:00-16:00

**Comment by azrikahar** [comment]
I believe an M2O field would be more suitable and flexible in such scenario, but understandably it can be difficult to understand the data modeling decisions made here without full context. I'll defer to @rijkvanzanten on whether this is a change we'd like to make (and/or any potential unforeseen pitfalls).

**Comment by sxsws** [comment]
First, OP is wrong. You have the text value in the value input and vice versa. It should be the other way around. Then technically, the values aren't duplicates as the values for each choice are different.

> I believe an M2O field would be more suitable and flexible in such scenario,

This. Personally I too tend to always use an m2o solution with a table being able to offer user editable choices and easy changes for the future. An `isActive` field can be used to filter them if I want to disable them at any stage. It would be a nightmare to have to remember where I've defined certain choices like this that 6+months down the line need to be changed. For clarity, I suffix all such collections with `Types`, so `paymentStatusTypes`, `organisationTypes` etc.

Also, in the case of business hours, I just use a repeater which an open time / close time structure.

**Comment by Pitcentr** [comment]
Thanks for the answer, in my example I create a field user mode (company) this field for each company is unique and different will create many duplicate M2O records, I created a field JSON and just in this case will store it in this form

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
